netpyne/sim/validator.py

```diff
@@ -0,0 +1,296 @@
+                    },
+                    Optional('ih'): {
+                        'aslope': float,
+                        'gbar': float,
+                        'ascale': float,
+                        'bslope': float,
+                        'bscale': float,
+                        'ashift': float
+                    }
+                },
+                Optional('threshold'): Or(int, float),
+                Optional('topol'): {
+                    Optional('parentSec'): str,
+                    Optional('childX'): Or(int, float),
+                    Optional('parentX'): Or(int, float)
+                },
+                Optional('pointps'): {
+                    str: {
+                        Optional('mod'): str,
+                        Optional('C'): Or(int, float),
+                        Optional('k'): Or(int, float),
+                        Optional('vr'): Or(int, float),
+                        Optional('vt'): Or(int, float),
+                        Optional('vpeak'): Or(int, float),
+                        Optional('a'): Or(int, float),
+                        Optional('b'): Or(int, float),
+                        Optional('c'): Or(int, float),
+                        Optional('d'): Or(int, float),
+                        Optional('celltype'): int,
+                        Optional('loc'): Or(int, float),
+                        Optional('vref'): str,
+                        Optional('synList'): [str]
+                    }
+                },
+                Optional('ions'): {
+                    Optional(str): {
+                        Optional('e'): Or(int, float),
+                        Optional('i'): Or(int, float),
+                        Optional('o'): Or(int, float)
+                    }
+                },
+                Optional('vinit'): Or(int, float),
+                Optional('weightNorm'): [float]
+            },
+        },
+        Optional('conds'): {
+            Optional('cellModel'): str,
+            Optional('cellType'): Or(str, [str])
+        },
+        Optional('globals'): {
+            str: Or(int, float)
+        },
+        Optional('secLists'): {
+            Optional(str): [str]
+        },
+        Optional('label'): str
+    }
+}
+
+population_spec = {
+    str: {
+        Optional('cellType'): str,
+        Optional('numCells'): int,
+        Optional('yRange'): [int],
+        Optional('ynormRange'): [float],
+        Optional('cellModel'): str,
+        Optional('density'): float,
+        Optional('originalFormat'): str,
+        Optional('pop'): str,
+        Optional('cellGids'): [int],
+        Optional('cellsList'): [
+            {
+                'cellLabel': int,
+                'x': float,
+                'y': float,
+                'z': float
+            }
+        ]
+    }
+}
+
+
+synaptic_spec = {
+    Optional(str): {
+        'mod': str, # Check to see if both of these strings are really valid
+        Optional('tau'): Or(int, float),
```

Comment:
these params are specific for a synaptic mod file, so cannot hard code here
ideally we would use netpyne calls (mechVarList() func) to inspect the parameters of each mod file dynamically and then validate